service-myplaces/src/main/java/fi/nls/oskari/myplaces/service/wfst/CategoriesWFSTRequestBuilder.java

```diff
@@ -0,0 +1,208 @@
+        writeTextElement(xsw, OSKARI, "stroke_linecap", category.getStroke_linecap());
+        writeTextElement(xsw, OSKARI, "stroke_dasharray", category.getStroke_dasharray());
+        writeTextElement(xsw, OSKARI, "border_linejoin", category.getBorder_linejoin());
+        writeTextElement(xsw, OSKARI, "border_dasharray", category.getBorder_dasharray());
+
+        xsw.writeEndElement(); // Close <feature:categories>
+        xsw.writeEndElement(); // Close <wfs:Insert>
+    }
+
+    private static void updateCategory(XMLStreamWriter xsw, MyPlaceCategory category)
+            throws XMLStreamException {
+        xsw.writeStartElement(WFS, "Update");
+        xsw.writeAttribute("typeName", TYPENAME_CATEGORIES);
+
+        writeProperty(xsw, "default", Boolean.toString(category.isDefault()));
+        writeProperty(xsw, "category_name", category.getCategory_name());
+        writeProperty(xsw, "stroke_width", category.getStroke_width());
+        writeProperty(xsw, "stroke_color", category.getStroke_color());
+        writeProperty(xsw, "fill_color", category.getFill_color());
+        writeProperty(xsw, "uuid", category.getUuid());
+        writeProperty(xsw, "dot_color", category.getDot_color());
+        writeProperty(xsw, "dot_size", category.getDot_size());
+        writeProperty(xsw, "border_width", category.getBorder_width());
+        writeProperty(xsw, "border_color", category.getBorder_color());
+        writeProperty(xsw, "publisher_name", category.getPublisher_name());
+        writeProperty(xsw, "dot_shape", category.getDot_shape());
+        writeProperty(xsw, "stroke_linejoin", category.getStroke_linejoin());
+        writeProperty(xsw, "fill_pattern", category.getFill_pattern());
+        writeProperty(xsw, "stroke_linecap", category.getStroke_linecap());
+        writeProperty(xsw, "stroke_dasharray", category.getStroke_dasharray());
+        writeProperty(xsw, "border_linejoin", category.getBorder_linejoin());
+        writeProperty(xsw, "border_dasharray", category.getBorder_dasharray());
+
+        writeFeatureIdFilter(xsw, prefixId(category.getId()));
+
+        xsw.writeEndElement(); // close <wfs:Update>
+    }
+
+    private static void deleteCategory(XMLStreamWriter xsw, long id)
+            throws XMLStreamException {
+        xsw.writeStartElement(WFS, "Delete");
+        xsw.writeAttribute("typeName", TYPENAME_CATEGORIES);
+        writeFeatureIdFilter(xsw, prefixId(id));
+        xsw.writeEndElement();
+    }
+
+    public static String prefixId(long id) {
+        return "categories." + id;
+    }
+
+    public static String[] prefixIds(long[] ids) {
+        final int n = ids.length;
+        String[] prefixed = new String[n];
+        for (int i = 0; i < n; i++) {
+            prefixed[i] = prefixId(ids[i]);
+        }
+        return prefixed;
+    }
+
+    public static long removePrefixFromId(String prefixed) {
+        return Long.parseLong(prefixed.substring("categories.".length()));
```

Comment:
Maybe check that prefixed.startsWith("categories.") and throw an exception about "unknown id" or something if not the expected prefix?